### ISSUE TITLE
Re-re do the API with respect to exceptions

### ISSUE THREAD

**Issue by dagit** [matterhorn-chat/mattermost-api#4]
The MM monad that we have is a good example of how someone might wrap up the API, but I'm now leaning towards that being an abstraction that we let the application programmer choose for themselves.

So here is the proposal:
- Take the MM monad out of base API and have the API explicitly take the auth token as a param and return types should be like `IO (Either String Foo)` or use `IO` exceptions like `IO Foo`.
- In the examples, provide an example MM monad that makes the API easier to use.

Basically, consumers of the API might want to use their own monad stack they might want to use mtl or monadLib instead of whatever we pick. So the point should be make it easy for people to wrap the base API up in whatever monad stack they want. We could even provide an example wrapped API as a separate package, but the API in this package should be the "raw" API.

**Comment by dagit** [comment]
Fixed in 2e90d95.
